torch/distributions/transforms.py

```diff
@@ -360,6 +361,29 @@ def log_abs_det_jacobian(self, x, y):
         return -(y.reciprocal() + (1 - y).reciprocal()).log()
 
 
+class TanhTransform(Transform):
+    r"""
+    Transform via the mapping :math:`y = \tanh(x)`.
+    """
+    domain = constraints.real
+    codomain = constraints.open_interval(-1.0, 1.0)
+    bijective = True
+    sign = +1
+    atanh = lambda x: 0.5*torch.log((1 + x)/(1 - x))
```

Comment:
This might be more stable using `log1p`. Also could you either define this as a file level helper `_atanh` or as a `staticmethod`, so that it has a name when it appears in a stack trace or profiling output?
```py
@staticmethod
def atanh(x):
    return 0.5 * (x.log1p() - (-x).log1p())
```